locomotor/__init__.py

```diff
@@ -1057,3 +1088,242 @@ def decorator(method):
         return functools.update_wrapper(fragment, method)
 
     return decorator(method) if method else decorator
+
+class minMaxVisitor(ast.NodeVisitor):
+    """
+    Ast node visitor to find the max and min line number of the body.
+    Max line number and min line number will be pass to find
+    the in_exprs and out_exprs.
+    """
+    def __init__(self, minValue, maxvalue):
+        self.minlineno = minValue
+        self.maxlineno = maxvalue
+
+    def generic_visit(self, node):
+
+        if hasattr(node, 'lineno'):
+            self.maxlineno = max(node.lineno, self.maxlineno)
+            self.minlineno = min(node.lineno, self.minlineno)
+        ast.NodeVisitor.generic_visit(self, node)
+
+
+
+def identifyArgumentType(taintFunction):
+    """ Identify type of argument """
+    for arg in taintFunction.func_ast.body[0].args.args:
+        if arg.annotation is not None:
+            if arg.annotation.id == 'list':
+                typeOfArgument = ast.List
+            elif arg.annotation.id == 'int' or arg.annotation.id == 'float' or arg.annotation.id == 'long':
+                typeOfArgument = ast.Num
+            elif arg.annotation.id == 'str':
+                typeOfArgument = ast.Str
+            taintFunction.variableTypes[arg.arg] = typeOfArgument
+
+    return taintFunction
+
+def identifyConvertLoops(method, redis_objs, taint):
```

Comment:
This method is very long and could benefit from being broken up into smaller pieces